R/md_document.R

```diff
@@ -74,17 +74,45 @@ md_document <- function(variant = "markdown_strict",
   }
 
   # return format
-  output_format(
-    knitr = knitr_options_html(fig_width, fig_height, fig_retina, FALSE, dev),
-    pandoc = pandoc_options(
-      to = variant,
-      from = from_rmarkdown(extensions = md_extensions),
-      args = args,
-      ext = ext,
-      lua_filters = if (number_sections) pkg_file_lua("number-sections.lua")
-    ),
-    clean_supporting = FALSE,
-    df_print = df_print,
-    post_processor = post_processor
-  )
+  if(ext == ".gfm") {
```

Comment:
The issue of ToC arises regardless of output markdown variant.
So I think making this conditional is not a good idea.
Also, it relies on file extension, which means people who want to output gfm with the `.md` extension will not benefit from the update.

Even if we implement this kind of conditional feature, I would say this if statement is too large.
If I were to, I would only make arguments to be conditional.

```
pre_processor = if (number_sections && ext == ".gfm") {
lua_filters = if (number_sections && (ext != ".gfm")) pkg_file_lua("number-sections.lua")
```

Reply:
I made the suggested changes. This will allow the programmer to make fine adjustments to the output format. I think it's important to be permissive about that.

For example, I have repositories where the table of contents are numbered and the links were broken in the generated .md file. Hence, I always needed to go to the Linux terminal and use pandoc to convert the file to gfm, since it was less work to do this than to edit the .md file.

In the end, I changed the extension .gfm to .md and had README.md working with the numbered links.

I know that allowing fine adjustments with the output format brings benefits that go beyond having a simple working table of contests. However, I also benefited from this flexibility in this situation.

There are people who may eventually want gfm for other uses. Allowing this does no harm to the code you already have and I think it is useful.

With regard to the links in the table of contents, I think that a Regex approach in .md will be dangerous, since there may always be a counter example that may not work. Having gfm for me is very useful.